flink-fs-tests/src/test/java/org/apache/flink/hdfstests/ContinuousFileProcessingTest.java

```diff
@@ -306,6 +311,21 @@ public int compare(String o1, String o2) {
 		}
 	}
 
+	private OneInputStreamOperatorTestHarness<TimestampedFileInputSplit, String> getTester(ContinuousFileReaderOperator<String> reader) throws Exception {
+		CompletableFuture<OneInputStreamOperatorTestHarness<TimestampedFileInputSplit, String>> future = new CompletableFuture<>();
+		new Thread(() -> {
+			try {
+				OneInputStreamOperatorTestHarness<TimestampedFileInputSplit, String> harness = new OneInputStreamOperatorTestHarness<>(reader);
```

Comment:
`Mailbox` loop must be executed by the same thread that created the `Harness` (because of thread check in the mailbox).
So only the whole method (create + run) can be moved to harness (and only to `OneInputStreamOperatorTestHarness` therefore), which I think is too specific.